modules/eks/main.tf

```diff
@@ -36,6 +37,14 @@ module "eks" {
         "materialize.cloud/disk" = "true"
         "workload"               = "materialize-instance"
       }
+
+      cloudinit_pre_nodeadm = [
```

Comment:
Using `cloudinit_pre_nodeadm` instead of user data bacause of:

 https://github.com/terraform-aws-modules/terraform-aws-eks/issues/3076#issuecomment-2187577220